### ISSUE TITLE
Increase in Latency In Past Week

### ISSUE THREAD

**Issue by idanadut** [wit-ai/wit#2492]
**Do you want to request a *feature*, report a *bug*, or ask a *question* about wit?**
BUG
**What is the current behavior?**
Substantial increase in "/message" requests latency
**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**
Make requests to the service and measure response time
**What is the expected behavior?**
100-200 ms request latency time
**If applicable, what is the App ID where you are experiencing this issue? If you do not provide this, we cannot help.**
337695227605625

The latency for the  "/message" requests is pretty high 500-2500ms or more at times. It has definitely increased in the past week. Is it not supposed to be around 100-200ms?
Can you please check also check specifically app Id 337695227605625. I'm not sure if it's a general issue or it could it be I'm hitting some quota or it happens in peak/rush hours of my app or of the wit service.
Thanks

**Comment by nishsinghal20** [comment]
Hi @idanadut! I had a chance to look at the latency history for your app, and going back up to 2 weeks, it seems that latency for your app (and for other apps as well) has been pretty consistent. There hasn't been any significant changes to your app's latency as far as I could see.
Let us know if you keep running into latency issues and we'll be happy to investigate more.

**Comment by idanadut** [comment]
Hi @nishsinghal20 , thanks for the inspection!
We did see some increase in the past 2-3 weeks, but it has improved a bit this week, but you know how latency issues are, it varies.
I have a different follow-up question then, 
(1) Is it possible I get higher latency in busy times (peak in requests)?
(2) I'm not sure what are the numbers you are currently seeing, but Is it possible to get the "/message" requests latency to an average of 100-200ms?
Thanks!

**Comment by serdalbayram** [comment]
the same problem for my app (1622020321212473). Latency is between 700 - 2000 msecs.

**Comment by nishsinghal20** [comment]
Hey @idanadut! 
> (1) Is it possible I get higher latency in busy times (peak in requests)?

Yes, it is possible that sometimes our systems are under heavy load and the latency can see occasional spikes.

> (2) I'm not sure what are the numbers you are currently seeing, but Is it possible to get the "/message" requests latency to an average of 100-200ms

Yes, definitely! We'll investigate how we can improve the latency.

@serdalbayram I looked into your app as well. While I do see latency numbers in that range for your app, there has been similar latency numbers for the past 2 weeks. 
But as I replied to @idanadut, we will definitely look into how we can improve these numbers. Thank you for reporting this! :)
